runtime/lua/vim/filetype.lua

```diff
@@ -863,6 +898,184 @@ local extension = {
       return "text"
     end
   end,
+  cmd = function (path, bufnr)
+    if getline(bufnr, 1):find("^/%*") then
+      return "rexx"
+    end
+    return "dosbatch"
+  end,
+  rul = function(path, bufnr)
+    if getline(bufnr, 1, 6):lower():find("installshield") then
+      return "ishd"
+    end
+    return "diva"
+  end,
+  cpy = function(path, bufnr)
+    if getline(bufnr, 1):find("^##") then
+      return "python"
+    end
+    return "cobol"
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*<!") then
+      return "dsl"
+    end
+    return "structurizr"
+  end,
+  edf = "edif",
+  edfi = "edif",
+  edo = "edif",
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*%(%s*edif>") then
+      return "edif"
+    end
+    return "clojure"
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find("<?%s*xml.*?>") then
+      return "xml"
+    end
+    return "smil"
+  end,
+  smi = function(path, bufnr)
+    if getline(bufnr, 1):find("<smil>") then
+      return "smil"
+    end
+    return "mib"
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    vim.fn["dist#ft#SetFileTypeSH"]("bash")
+  end,
+  pm = function(path, bufnr)
+    if getline(bufnr, 1):find("XPM2") then
+      return "xpm2"
+    elseif getline(bufnr, 1):find("XPM") then
+      return "xpm"
+    else
+      return "perl"
+    end
+  end,
+  me = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t")
+    if filename ~= "read.me" and filename ~= "click.me" then
+      return "nroff"
+    end
+  end,
+  reg = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find("^REGEDIT[0-9]*%s*$") or line:find("^Windows Registry Editor Version %d*%.%d*%s*$") then
+      return "registry"
+    end
+  end,
+  decl = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
```

Comment:
This does not do what you are intending to do: `^` will only match the start of all concatenated lines (i.e. always line 1). There needs to be a loop iterating over the lines here. The same applies for all following occurences where more than one line is compared. (see https://github.com/neovim/neovim/blob/95b7851f620928b6a93dc0198be31959b16c53d9/runtime/lua/vim/filetype/detect.lua#L112 for an example). Maybe we should use [`getlines`](https://github.com/neovim/neovim/blob/95b7851f620928b6a93dc0198be31959b16c53d9/runtime/lua/vim/filetype/detect.lua#L4) from `detect.lua` here as well?

Reply:
Or -- better -- move this check to `detect.lua`? Rule of thumb (for now) could be that any check that looks at more than the first line belongs in `detect.lua`.